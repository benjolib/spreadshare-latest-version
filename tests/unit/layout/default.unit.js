// @flow

import DefaultLayout from '~/layouts/default'
import { shallowMount } from '@vue/test-utils'

describe('defaultLayout', () => {
  const wrapper = shallowMount(DefaultLayout, {
    stubs: ['nuxt'],
  })

  describe('unit tests', () => {
    test('is a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper.is(DefaultLayout)).toBeTruthy()
    })

    test('should have app id', () => {
      const div = wrapper.find('div')
      expect(div.attributes().id).toBe('app')
    })

    test('should have navBar', () => {
      expect(wrapper.find('navbar-stub').exists()).toBeTruthy()
    })

    test('should have nuxt element', () => {
      const nuxt = wrapper.find('nuxt-stub')
      expect(nuxt.attributes().role).toBe('main')
      expect(nuxt.attributes()['nuxt-child-key']).toBe('none')
    })
  })

  describe('snapshot tests', () => {
    test('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
