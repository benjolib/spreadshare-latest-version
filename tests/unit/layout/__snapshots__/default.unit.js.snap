// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultLayout snapshot tests should match the snapshot 1`] = `
<div
  id="app"
>
  <navbar-stub />
   
  <nuxt-stub
    nuxt-child-key="none"
    role="main"
  />
</div>
`;
